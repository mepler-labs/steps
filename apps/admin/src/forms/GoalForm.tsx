import Button from 'atoms/Buttons/Button';
import Panel from 'atoms/Panel';
import Textarea from 'components/Form/Textarea';
import { Goal } from 'components/Goals/types';
import { Flex } from 'grid-styled';
import React from 'react';
import { Form } from 'react-final-form';

interface Props {
  onSubmit: (goal: Goal) => void;
  goal?: Goal;
}

const GoalForm: React.SFC<Props> = ({ onSubmit, goal }) => (
  <Form
    initialValues={{ goal: goal && goal.text }}
    onSubmit={onSubmit}
    render={({ handleSubmit }) => (
      <form onSubmit={handleSubmit}>
        <h2>{goal ? 'Edit Goal' : 'Add New Goal'}</h2>
        <Panel>
          <Textarea name="goal" placeholder="My goal is to..." label="goal" />
          <Flex justifyContent="center">
            <Button>Save</Button>
          </Flex>
        </Panel>
      </form>
    )}
  />
);

export default GoalForm;
