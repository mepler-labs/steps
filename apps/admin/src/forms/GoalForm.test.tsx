import { shallow } from 'enzyme';
import 'jest';
import React from 'react';
import { Form } from 'react-final-form';
import GoalForm from './GoalForm';

describe('GoalForm.tsx', () => {
  it('renders correctly', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<GoalForm onSubmit={onSubmit} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('submits the form', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<GoalForm onSubmit={onSubmit} />);

    wrapper.simulate('submit', { some: 'data' });

    expect(onSubmit).toHaveBeenCalledWith({ some: 'data' });
  });

  it('displays Add New Goal if no goal exists', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<GoalForm onSubmit={onSubmit} />);

    const header = wrapper.dive().find('h2');

    expect(header.text()).toEqual('Add New Goal');
  });

  it('displays Edit Goal if goal exists', () => {
    const onSubmit = jest.fn();
    const goal = { text: 'My goal' };

    const wrapper = shallow(<GoalForm onSubmit={onSubmit} goal={goal} />);

    const header = wrapper.dive().find('h2');

    expect(header.text()).toEqual('Edit Goal');
  });

  it('displays goal text if goal exists', () => {
    const onSubmit = jest.fn();
    const goal = { text: 'My goal' };

    const wrapper = shallow(<GoalForm onSubmit={onSubmit} goal={goal} />);

    const form = wrapper.find(Form);

    expect(form.props().initialValues).toEqual({ goal: 'My goal' });
  });
});
