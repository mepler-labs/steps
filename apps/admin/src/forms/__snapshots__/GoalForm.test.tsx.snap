// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalForm.tsx renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add New Goal
  </h2>
  <Panel>
    <Textarea
      label="goal"
      name="goal"
      placeholder="My goal is to..."
    />
    <Flex
      justifyContent="center"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <Button>
        Save
      </Button>
    </Flex>
  </Panel>
</form>
`;
